Sources/FoundationEssentials/Error/ErrorCodes+Win32.swift

```diff
@@ -0,0 +1,33 @@
+//===----------------------------------------------------------------------===//
+ //
+ // This source file is part of the Swift.org open source project
+ //
+ // Copyright (c) 2022 Apple Inc. and the Swift project authors
+ // Licensed under Apache License v2.0 with Runtime Library Exception
+ //
+ // See https://swift.org/LICENSE.txt for license information
+ //
+ //===----------------------------------------------------------------------===//
+
+#if os(Windows)
+import WinSDK
+
+public struct Win32Error: Error {
```

Comment:
Since this is a new `public` type, we might want to run this through review. @itingliu thoughts?

Reply:
That sounds like a good plan to me